osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/openstack/sfc/CheckServiceFunctionChainMetaTask.java

```
@@ -0,0 +1,157 @@
+					this.securityGroup.getServiceFunctionChain().getVirtualSystems());
+
+			try (SdnRedirectionApi sdnApi = this.apiFactoryService
+					.createNetworkRedirectionApi(this.securityGroup.getVirtualizationConnector())) {
+
+				List<NetworkElement> sfcPortPairGroups = sdnApi
+						.getNetworkElements(new NetworkElementImpl(this.securityGroup.getNetworkElementId()));
+
+				if (!this.portPairGroups.equals(sfcPortPairGroups)) {
+					this.tg.appendTask(
+							this.updateServiceFunctionChainTask.create(this.securityGroup, this.portPairGroups));
+				}
+			}
+
+		} else if (this.securityGroup.getNetworkElementId() != null && this.securityGroup.getServiceFunctionChain() == null) {
+			// if SFC not binded and network element not null -> delete PC
+			this.tg.appendTask(this.deleteServiceFunctionChainTask.create(this.securityGroup));
+
+		}
+	}
+
+	static List<NetworkElement> getPortPairGroupWithSameProjectId(EntityManager em, String projectId,
+			List<VirtualSystem> vsList) throws VmidcBrokerValidationException {
+
+		List<NetworkElement> portPairGroups = new ArrayList<>();
+
+		for (VirtualSystem vs : vsList) {
+			if (vs.getDeploymentSpecs().isEmpty()) {
+				throw new VmidcBrokerValidationException(
+						String.format("No deployment found for virtual system '%s'.", vs.getName()));
+			}
+
+			// Currently, We have no information on OSC SG about the OST region. SFC do not support port chain with multiple region deployment.
+			// We cannot create Port Chain of VNF Deployment's belonging to different region. If we find the VS with multiple DS on different region throw an exception.
+			List<DeploymentSpec> dsList = DeploymentSpecEntityMgr
+					.findDeploymentSpecsByVirtualSystemProjectWithDefaultRegionOne(em, vs, projectId);
+			if (dsList.size() > 1) {
+				throw new VmidcBrokerValidationException(String
+						.format("Multiple deployments found for vs '%s' belonging to the same project.", vs.getName()));
+			}
+
+			if (!dsList.isEmpty()) {
```

if dslist is empty we need to throw an exception right specifying there are no deployments for the binded chain.

---

We already have it.

---

can you point me to the code?

---

can you point me to the code?

I see if the ds's PortGroupId is set to null/empty we throw the exception...but not for the case when there are 0 ds'es

---

https://github.com/larkinscarvalho/osc-core/blob/sfc/osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/openstack/sfc/CheckServiceFunctionChainMetaTask.java#L115

---

We check for each vs if we have deployment specs

---

Having a DS is not enough though. you have to have a "matching" ds. I would remove that check and use the list you get from findDeploymentSpecsByVirtualSystemProjectWithDefaultRegionOne